compiler/rustc_resolve/src/ident.rs

```diff
@@ -1522,6 +1531,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     let maybe_assoc = opt_ns != Some(MacroNS) && PathSource::Type.is_expected(res);
```

Comment:
I think you can move the whole
```rust
        for error in &mut self.privacy_errors[privacy_errors_len..] {
            error.outermost_res = Some((res, ident));
        }
```
loop to this point (right after `let res = ...`).

Reply:
Wouldn't this mean that we're doing `O(privacy_errors_len * path_segments_len)` operations? The first condition, `if let Some(net_module) = binding.module()` will fallthrough and move to the next path segment, iterating over the privacy errors and setting them again (until we return).

Reply:
`self.privacy_errors[privacy_errors_len..].len()` is zero during successful compilation, so we don't slow down the good path.

And even on a bad path, where perf doesn't matter much, there should be about `1 * path_segments_len` operations, which is nothing.